### ISSUE TITLE
Add an example for adding a reverse record

### ISSUE THREAD

**Issue by pezhore** [joeig/go-powerdns#74]
I've been brute forcing an effort today to figure out how to make a reverse record (PTR), and I just cannot figure it out. It would be good to have an example for reverse records.

Here's some code that I've used to try and run through all the options I can think of. Warning, it's long...
<details>
<summary> Go code with error generating code </summary>

```
// IP addr: 10.1.2.3
ip := "10.1.2.3"
fqdn := "host.example.com"
zone := "10.in-addr.arpa."
record := "3.2.1"
rip := "3.2.1.10"

err = pdns.Records.Add(ctx, zone, fqdn, powerdns.RRTypePTR, 60, []string{fmt.Sprintf("%s.in-addr.arpa.", ip)})
if err != nil {
log.Warn().Msgf("errored at 1 %v", err)
}

// name is full, non-reversed ip, content is fqdn
err = pdns.Records.Add(ctx, zone, fmt.Sprintf("%s.in-addr.arpa.", ip), powerdns.RRTypePTR, 60, []string{fqdn})
if err != nil {
log.Warn().Msgf("errored at 2 %v", err)
}

// name is fqdn, record is our truncated, reversed ip
err = pdns.Records.Add(ctx, zone, fqdn, powerdns.RRTypePTR, 60, []string{fmt.Sprintf("%s.in-addr.arpa.", record)})
if err != nil {
log.Warn().Msgf("errored at 3 %v", err)
}

// name is truncated, reversed ip record is our fqdn
err = pdns.Records.Add(ctx, zone, fmt.Sprintf("%s.in-addr.arpa.", record), powerdns.RRTypePTR, 60, []string{fqdn})
if err != nil {
log.Warn().Msgf("errored at 4 %v", err)
}

// name is fqdn, record is full, reversed ip
err = pdns.Records.Add(ctx, zone, fqdn, powerdns.RRTypePTR, 60, []string{fmt.Sprintf("%s.in-addr.arpa.", rip)})
if err != nil {
log.Warn().Msgf("errored at 5 %v", err)
}

// name is full, reversed ip, record is fqdn
err = pdns.Records.Add(ctx, zone, fmt.Sprintf("%s.in-addr.arpa.", rip), powerdns.RRTypePTR, 60, []string{fqdn})
if err != nil {
log.Warn().Msgf("errored at 6 %v", err)
}

// Same as above but without the ".in-addr.arpa."
err = pdns.Records.Add(ctx, zone, fqdn, powerdns.RRTypePTR, 60, []string{ip})
if err != nil {
log.Warn().Msgf("errored at 7 %v", err)
}

// name is full, non-reversed ip, content is fqdn
err = pdns.Records.Add(ctx, zone, ip, powerdns.RRTypePTR, 60, []string{fqdn})
if err != nil {
log.Warn().Msgf("errored at 8 %v", err)
}

// name is fqdn, record is our truncated, reversed ip
err = pdns.Records.Add(ctx, zone, fqdn, powerdns.RRTypePTR, 60, []string{record})
if err != nil {
log.Warn().Msgf("errored at 9 %v", err)
}

// name is truncated, reversed ip record is our fqdn
err = pdns.Records.Add(ctx, zone, record, powerdns.RRTypePTR, 60, []string{fqdn})
if err != nil {
log.Warn().Msgf("errored at 10 %v", err)
}

// name is fqdn, record is full, reversed ip
err = pdns.Records.Add(ctx, zone, fqdn, powerdns.RRTypePTR, 60, []string{rip})
if err != nil {
log.Warn().Msgf("errored at 11 %v", err)
}

// name is full, reversed ip, record is fqdn
err = pdns.Records.Add(ctx, zone, rip, powerdns.RRTypePTR, 60, []string{fqdn})
if err != nil {
log.Warn().Msgf("errored at 12 %v", err)
}
}
```
</details>

That code when run generates the following:

```
5:43PM WRN errored at 1 RRset host.example.com. IN PTR: Name is out of zone
5:43PM WRN errored at 2 RRset 10.1.2.3.in-addr.arpa. IN PTR: Name is out of zone
5:43PM WRN errored at 3 RRset host.example.com. IN PTR: Name is out of zone
5:43PM WRN errored at 4 RRset 3.2.1.in-addr.arpa. IN PTR: Name is out of zone
5:43PM WRN errored at 5 RRset host.example.com. IN PTR: Name is out of zone
5:43PM WRN errored at 6 Record 3.2.1.10.in-addr.arpa./PTR 'host.example.com': Not in expected format (parsed as 'host.example.com.')
5:43PM WRN errored at 7 RRset host.example.com. IN PTR: Name is out of zone
5:43PM WRN errored at 8 RRset 10.1.2.3. IN PTR: Name is out of zone
5:43PM WRN errored at 9 RRset host.example.com. IN PTR: Name is out of zone
5:43PM WRN errored at 10 RRset 3.2.1. IN PTR: Name is out of zone
5:43PM WRN errored at 11 RRset host.example.com. IN PTR: Name is out of zone
5:43PM WRN errored at 12 RRset 3.2.1.10. IN PTR: Name is out of zone
```

In other words, I've tried the following combinations (along with everything below, but with `.in-addr.arpa.` added to the IP addresses:

| Name | Content |
|--- |---|
| fqdn | full, normal ordered IP |
| full, normal orderd ip | fqdn |
| fqdn | reversed, truncated IP |
| reversed, truncated ip | fqdn |
|fqdn | reversed, not-truncated ip|
| reversed, not-truncated ip| fqdn|

I can't think of any other options to try here, so I must be missing something (probably something obvious).

I would expect either the record name or the value to be something like `3.2.1.in-addr.arpa.` or `3.2.1` as that's how things work for reverse records.

Any ideas? I'm happy to do a PR with a working example once I've discovered the proper call.

**Comment by pezhore** [comment]
And of course after just submitting this, I found the correct combination... for posterity. (note the fully qualified host in the content).

```
err = pdns.Records.Add(ctx, "10.in-addr.arpa", "3.2.1.in-addr.arpa", powerdns.RRTypePTR, 60, []string{"host.example.com."})
```

As promised, I'll get some example changes written up and submit a PR.

Thanks to the developer for this project, it's a life saver!
